### ISSUE TITLE
about m3 finetune

### ISSUE THREAD

**Issue by jhyeom1545** [FlagOpen/FlagEmbedding#1074]
Hi, I have some question about m3 model finetune

1. I don't understand the role of What is fix_encode True and Unified_finetuning True. Can you explain?

2. About small dataset and large dataset의 learning rate
- Model: bge-m3: For 500 levels of fine tuning, I think the running rate should be higher. ex) 1e-5
- Model: Unsupervised: If it's around 500,000, I think we should lower the running rate. e) 5e-6
Will this be the recommended args of fine tuning?

3. About fine tuning
- Is it better recommended to use https://github.com/FlagOpen/FlagEmbedding/tree/master/examples/finetune to fine tune dense backgrounds? Or, if --unified_defetuning True is set to False, will only dense vectors be learned, I mainly use dense vectors, and I wonder if using dense vectors alone can make learning progress more efficiently (memory efficient, faster training).

Thank you

**Comment by hanhainebula** [comment]
Hello, @jhyeom1545!
1. When `fix_encoder` is set to `True`, only sparse(sparse_linear.pt) and multi-vector(colbert_linear.pt) methods are finetuned, and the parameters of dense method are fixed. When `unified_finetuning` is set to `True`, all three methods will be finetuned. Therefore, if you only want to finetune dense method, set `fix_encoder=False` and `unfied_finetuning=False`. If you only want to finetune sparse and multi-vector methods, set `fix_encoder=True` and `unified_finetuning=True`. If you want to finetune all three methods, set `fix_encoder=False` and `unfied_finetuning=True`.
2. Generally speaking, the learning rate value depends on the size of model you want to finetune (larger model, smaller lr), and the size of datasets you want to use (smaller dataset, smaller lr). The values you want to use (1e-5 and 5e-6) look good. You can also conduct some comparative experiments to determine the optimal learning rate.
3. If you set `unified_finetuning=False`, then only dense method will be finetuned. If you mainly use dense vectors, then only finetuning dense method will be a good choice, which will be more efficient than unified finetuning. Use https://github.com/FlagOpen/FlagEmbedding/tree/master/examples/finetune or https://github.com/FlagOpen/FlagEmbedding/tree/master/examples/unified_finetune are both okay.

**Comment by jhyeom1545** [comment]
@hanhainebula Thank you for your reply!
